src/stream/src/executor/backfill/cdc/cdc_backfill.rs

```diff
@@ -334,8 +343,16 @@ impl<S: StateStore> CdcBackfillExecutor<S> {
                                         last_binlog_offset = consumed_binlog_offset.clone();
                                     }
 
-                                    // seal current epoch even though there is no data
+                                    // update and persist backfill state
+                                    state_impl
+                                        .mutate_state(CdcStateItem::new(
+                                            false,
+                                            last_binlog_offset.clone(),
+                                            total_snapshot_processed_rows as _,
```

Comment:
IIUC, `last_binlog_offset` is not used after restoring the state and we don't store the pk position for mysql snapshot read. Are we planning to support resume backfill on the exact pk after recovery? If yes, will it be incompatible in the future?

Reply:
I updated the PR to support recoverable backfill.